.github/workflows/docker.yaml

```diff
@@ -28,6 +33,15 @@ jobs:
           registry: ghcr.io
           username: ${{ github.repository_owner }}
           password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Check apt updates
+        if: ${{ github.event_name == 'schedule' || github.event_name == 'workflow_dispatch' }}
+        run: |
+            apt-get update -o Dir::Etc::sourcelist="${SOURCELIST}"
+            apt-get --simulate upgrade -o Dir::Etc::sourcelist="${SOURCELIST}" \
+            | grep "^0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.$" \
+            && echo "::set-output name=trigger::false" \
+            || echo "::set-output name=trigger::true"
+            echo "::set-output name=no_cache::true"
```

Comment:
@ruffsl, thinking deeper about this code snippet from nav2, I think it doesn't make sense: 
This code actually checks for system updates in the OS _running the job_, but not for updates _within_ the docker image.

My suggestion is to disable caching for scheduled jobs and thus fetch new updates once a week. See https://github.com/vatanaksoytezer/moveit2/pull/17

Reply:
@rhaschke I think it actually runs on the previous CI image, see https://github.com/ros-planning/navigation2/blob/main/.github/workflows/update_ci_image.yaml#L38-L39.

Reply:
Ah, you are right. I missed the reference to the container! However, you missed copying this from `nav2` :wink:

Reply:
Yep you are right, I actually accidentally pushed that bit prematurely. I am working on something manageable to check the build triggers